### ISSUE TITLE
avoiding wayland

### ISSUE THREAD

**Issue by theotrst** [meganz/MEGAsync#904]
Hey,
I have recently reinstalled a clean arch distro (endeavourOS without DE) with hyprland/waybar. I have tried to install megasync (as megasync itself, and other versions as "-bin",  "-nopdfium", the .pkg.tar.zts package for arch). Installation is ok, but when I launch it I got 
```
$ megasync
Avoiding wayland
```
no other info are provided (I found around similar issues with problems of environment variables).
Is there someone with an idea to solve the problem?

thanks

**Comment by khinch** [comment]
It's possible this is related to Hyprland or lack of DE. I am running NixOS and Mega runs OK on Gnome/Wayland.

**Comment by SSouper** [comment]
Same issue here, fresh install of Fedora 39 running GNOME

**Comment by theotrst** [comment]
> Edit: I just noticed that if I run it from the terminal, I also get the same message, but it still starts fine. I wonder if this error is not the root cause of your problem.
> 
Thank you. I tried this and I got: 

```
[rdw@rdw ~]$ megasync
Avoiding wayland
[rdw@rdw ~]$ sudo megasync
[sudo] password di rdw: 
Error for "xrdb -query" command: "Authorization required, but no authorization protocol specified\n\nxrdb: Resource temporarily unavailable\nxrdb: Can't open display ':0'\n"
Authorization required, but no authorization protocol specified

qt.qpa.xcb: could not connect to display :0
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, xcb.

Annullato
[rdw@rdw ~]$ 
```
It seems it is a problem of environment variables. Do you have the same issue?

**Comment by rautamiekka** [comment]
Adding `sudo` changes to the `root` user, and `root` doesn't have X running (or something else acting like one), ergo there's no `DISPLAY` env var to use, hence inability to connect to display.

**Comment by theotrst** [comment]
yes, right. I forgot that. It makes sense. But anyway I do not understand why when I run megasync with my environment, it does not work.

**Comment by GixoXYZ** [comment]
The issue is that the login screen is not shown in Hyprland, but if you login to megasync on another DE, it'll start working just fine. It's just annoying being forced to do so. I thought of using megasyc-cli for loging in, but haven't tried it.

**Comment by hongoneko** [comment]
I run 'sudo megasync' and get the following message. QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'

The sync worked first time but it tagged all the downloaded files under sudo and not I had to use nautilus to access and open up the permission to other groups. 

Megasync now doesn't realize I've added/modified files on the local folder so the sync back up doesn't work..

What a mess!

**Comment by rautamiekka** [comment]
Don't ever run anything as `root` that you don't need to, of which MEGAsync ain't one !!

**Comment by hongoneko** [comment]
how do I install megasync then?
root is the only way it runs now... see thread

**Comment by hongoneko** [comment]
Solved ! I moved to KDE !

**Comment by theotrst** [comment]
Actually, I noticed it works in other DE, but -as said by GixoXYZ- it is quite annoying to have to use another DE only for synchronizing megasync.

**Comment by zadjadr** [comment]
I have just switched to using the `mega-cmd` tool instead of the GUI.

```shell
mega-cmd

> login myemail@example.com
# It will promt you for password and 2FA code
> sync /local/path /mega-remote/path
# e.g. /BACKUP/MEGA / 
```

You can add `mega-sync` to be started after boot.

https://github.com/meganz/MEGAcmd/blob/master/UserGuide.md#sync-example

**Comment by Salvodif** [comment]
> s in other DE, but -as said by GixoXYZ- it is quite annoying to have to use

same issue here on CachyOS + KDE
but everything ok on Endeavour + KDE
and on Manjaro + KDE....

thoughts?

**Comment by kryzar** [comment]
> Don't ever run anything as `root` that you don't need to, of which MEGAsync ain't one !!

My god, yes. Thank you. This would NOT be a good solution to this problem.

**Comment by Armadillidiid** [comment]
I'm experiencing this issue on Hyprland too. It launches fine despite the wayland warning but the app immediately minimizes to the system tray upon clicking any button. I can't login because of this.

**Comment by FilMar** [comment]
> I'm experiencing this issue on Hyprland too. It launches fine despite the wayland warning but the app immediately minimizes to the system tray upon clicking any button. I can't login because of this.

I have the same problem, any ideas to solve it?

**Comment by Armadillidiid** [comment]
> > I'm experiencing this issue on Hyprland too. It launches fine despite the wayland warning but the app immediately minimizes to the system tray upon clicking any button. I can't login because of this.
> 
> I have the same problem, any ideas to solve it?

I couldn't get it to work and resolved to using their cli `mega-cmd`. It's actually not bad, you can doing everything you can on the app as far I can tell.

**Comment by FilMar** [comment]
> > > I'm experiencing this issue on Hyprland too. It launches fine despite the wayland warning but the app immediately minimizes to the system tray upon clicking any button. I can't login because of this.
> > 
> > 
> > I have the same problem, any ideas to solve it?
> 
> I couldn't get it to work and resolved to using their cli `mega-cmd`. It's actually not bad, you can doing everything you can on the app as far as I can tell.

At the beginning i used mega-cmd, but from what i see it not use .megaignore file and i need it because i need to exclude all .* files and folder except one.
You know a way to do that with mega-cmd?

**Comment by Armadillidiid** [comment]
Frankly I don't, sorry.

**Comment by theotrst** [comment]
apparently, it works for me on my Arch on Hyprland, 

uncommenting:
# QT_QPA_PLATFORMTHEME=qt6ct

and adding:
QT_SCALE_FACTOR=1 
QT_QPA_PLATFORM=xcb megasync

as described [here](https://github.com/meganz/MEGAsync/issues/710)

**Comment by voice-hearing** [comment]
# Solution for Fedora 41 (I suspect it might work on any other wayland distro)

I got Fedora sway atomic running in 3 laptops and can confirm that megasync to this date 2025/03/08 is avoiding wayland.

## Installing megasync
distrobox create -n fedora
distrobox enter fedora
sudo dnf install megasync
distrobox-export --app megasync 

### megasync can be openened from terminal or menu

Cheers Faust :-)

**Comment by ThLekkas** [comment]
I was facing the same issues with MEGAsync and Hyprland on arch linux but after trying everything on every forum I noticed that after i click on the "Sign In" button a small dot appeared on the middle of the screen and when i went to resize the window (the dot) then it appeared!
I hope this may help someone
Cheers!
